sycl/include/sycl/queue.hpp

```diff
@@ -3699,6 +3768,77 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
                                   TlsCodeLocCapture.isToplevel());
   }
 
+  template <typename KernelName = detail::auto_name, typename PropertiesT,
+            typename KernelType, int Dims>
+  event
+  submit_kernel_direct_with_event(PropertiesT Props, nd_range<Dims> Range,
+                                  const KernelType &KernelFunc,
+                                  const detail::code_location &CodeLoc =
+                                      detail::code_location::current()) const {
+    (void)Props;
+    detail::tls_code_loc_t TlsCodeLocCapture(CodeLoc);
+    std::shared_ptr<detail::HostKernelBase> HostKernel;
+    detail::DeviceKernelInfo *DeviceKernelInfoPtr = nullptr;
+
+    using NameT =
+        typename detail::get_kernel_name_t<KernelName, KernelType>::name;
+    using LambdaArgType =
+        sycl::detail::lambda_arg_type<KernelType, nd_item<Dims>>;
+    static_assert(
+        std::is_convertible_v<sycl::nd_item<Dims>, LambdaArgType>,
+        "Kernel argument of a sycl::parallel_for with sycl::nd_range "
+        "must be either sycl::nd_item or be convertible from sycl::nd_item");
+    using TransformedArgType = sycl::nd_item<Dims>;
+
+    HostKernel.reset(
+        new detail::HostKernel<KernelType, TransformedArgType, Dims>(
+            KernelFunc));
+
+    DeviceKernelInfoPtr = &detail::getDeviceKernelInfo<NameT>();
```

Comment:
Just declare the `DeviceKernelInfoPtr` here and assign its value here, and remove the declaration from the L3781.